modules/packages/LinearAlgebra.chpl

```diff
@@ -2490,6 +2490,372 @@ proc cosm(A: []) throws {
   }
 }
 
+// Gets sums along specified axes. Method for Rectangular matrix.
+proc absSum(A: [?D], param axis=0) where !isCSDom(D) {
+  if (axis != 0 || axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  if axis == 0 then
+    // axis=0 sums along column
+    return forall j in D.dim(1-axis) do (+ reduce abs(A[D.dim(axis), j..j]));
+  else if axis == 1 then
+    // axis=1 sums along row
+    return forall i in D.dim(1-axis) do (+ reduce abs(A[i..i, D.dim(axis)]));
+}
+
+// Gets sums along specified axes. Method for CSR matrix.
+proc absSum(A: [?ADom], param axis=0) where isCSDom(ADom) {
+  if (axis != 0 || axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[j] += abs(A[i,j]);
+    }
+    return b;
+  }
+  else if axis == 1 {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[i] += abs(A[i,j]);
+    }
+    return b;
+  }
+}
+
+// Gets max along specified axes. Method for CSR matrix.
+proc max(A: [?ADom], param axis=0) where isCSDom(ADom) {
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
```

Comment:
Again, I think this is a race on b. I think you need a `with (max reduce b)` here as well.

Reply:
Ah! I had this on the checklist before pushing. I missed it! I'll add this.